vault-core/src/main/java/org/apache/jackrabbit/vault/fs/impl/io/DocViewImporter.java

```diff
@@ -989,8 +989,17 @@ private void removeReferences(@NotNull Node node) throws ReferentialIntegrityExc
         VersioningState vs = new VersioningState(stack, node);
         Node updatedNode = null;
         Optional<String> identifier = ni.getIdentifier();
+
+        String nodeUUID = null;
+        try {
+            nodeUUID = node.getUUID();
```

Comment:
> Deprecated. As of JCR 2.0, [getIdentifier()](https://developer.adobe.com/experience-manager/reference-materials/spec/javax.jcr/javadocs/jcr-2.0/javax/jcr/Node.html#getIdentifier()) should be used instead.

(https://developer.adobe.com/experience-manager/reference-materials/spec/javax.jcr/javadocs/jcr-2.0/javax/jcr/Node.html#getUUID())

I am not understanding how that makes a diference because this is all only relevant if the node supposed to be update/replaced has a uuid and IMHO the return value should be the same for getUUID and getIdentifier in case the node is referenceable right?

Reply:
Yes, but this is the simplest way to find out whether the node is referenceable. The other things I tried all require a lot of work with checking node types.

FWIW, the deprecation of `getUUID()` IMHO was incorrect, because these two have different semantics for non-referenceable nodes.

Reply:
checking if the property jcr:uuid is protected should be pretty straightforward

Reply:
That sounds good, I'll give that a try.

Reply:
so

hasProperty -> getProperty -> getDefinition -> isProtected

?